src/librustc_lint/non_ascii_idents.rs

```diff
@@ -13,9 +14,50 @@ declare_lint! {
     "detects uncommon Unicode codepoints in identifiers"
 }
 
-declare_lint_pass!(NonAsciiIdents => [NON_ASCII_IDENTS, UNCOMMON_CODEPOINTS]);
+declare_lint! {
+    pub CONFUSABLE_IDENTS,
+    Warn,
+    "detects uncommon Unicode codepoints in identifiers"
+}
+
+declare_lint_pass!(NonAsciiIdents => [NON_ASCII_IDENTS, UNCOMMON_CODEPOINTS, CONFUSABLE_IDENTS]);
+
+fn calc_skeleton(symbol_str: &str) -> String {
+    use unicode_security::confusable_detection::skeleton;
+    skeleton(symbol_str).collect()
```

Comment:
Seems like we should be able to at least reuse a single String here, just extending it repeatedly, avoiding an allocation for ~every identifier.

Reply:
Seems entry API is taking ownership... But i think i can use [raw_entry API](https://doc.rust-lang.org/nightly/std/collections/struct.HashMap.html#method.raw_entry_mut) to do this. Never used it before, i will read the docs and give it a try.